astroid/inference_tip.py

```diff
@@ -3,16 +3,20 @@
 
 """Transform utilities (filters and decorator)"""
 
-import typing
+
+from typing import Any, Callable, Dict, Iterator, List, Optional, Tuple, Type, Union
 
 import wrapt
 
-from astroid.exceptions import InferenceOverwriteError
-from astroid.nodes import NodeNG
+from astroid import bases, nodes, util
```

Comment:
There are some circular import in astroid, between files and between packages, I think importing only what's necessary and not whole package will help to understand what is happening in the code if we want to one day handle the issue.

Reply:
On the other hand, importing submodules avoid such import errors as we will get notified of those patterns sooner.

But I'll revert if you want me to!

Reply:
@cdce8p do you have an opinion ? It's been a long time since I tried to fix those circular imports, all I remember is the pain 😄